src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,438 @@
+                } else {
+                    try {
+                        listCount = createTraditionalTask(input, listCount, tasks, false);
+                    } catch (InvalidCommandException e) {
+                        e.printStackTrace();
+                    }
+                }
+            }
+        }
+    }
+
+    public static void displayWelcomeMessage() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+        System.out.println("What can I do for you?");
+    }
+
+    public static int deleteTask(String input, ArrayList<Task> tasks, int listCount) {
+        final int DELETE_OFFSET = 7;
+        final int ARRAY_OFFSET = 1;
+        final int TODO = 3;
+        final int EVENT = 2;
+        final int DEADLINE = 1;
+        //Traditional tasks are tasks specified in Level-2
+        final int TRADITIONAL_TASK = 0;
+        int lastNrPosition = input.length();
+        String sub = input.substring(DELETE_OFFSET, lastNrPosition);
+        int i = 0;
+        String status;
+
+        try {
+            i = Integer.parseInt(sub) - ARRAY_OFFSET;
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Invalid task number. No items are deleted.");
+        } catch (NumberFormatException e) {
+            System.out.println("Please enter a number!");
+        }
+        try {
+            int taskType = tasks.get(i).getTaskType();
+            status = tasks.get(i).getStatusIcon();
+            System.out.println("Noted. I've removed this task: ");
+            if (taskType == TRADITIONAL_TASK) {
+                System.out.println("  [" + status + "] " + tasks.get(i).getDescription());
+            } else if (taskType == DEADLINE) {
+                String temp = tasks.get(i).getTime();
+                System.out.println("  [D][" + status + "] " + tasks.get(i).getDescription() + "(by:" + temp + ")");
+            } else if (taskType == EVENT) {
+                String temp = tasks.get(i).getTime();
+                System.out.println("  [E][" + status + "] " + tasks.get(i).getDescription() + "(on:" + temp + ")");
+            } else if (taskType == TODO) {
+                System.out.println("  [T][" + status + "] " + tasks.get(i).getDescription());
+            }
+            tasks.remove(i);
+            listCount--;
+            return listCount;
+        }
+        catch (IndexOutOfBoundsException e) {
+            System.out.println("Invalid task number. No items are deleted.");
+        }
+        return listCount;
+    }
+
+    public static void flagAsDone(String input, ArrayList<Task> tasks) {
+        final int IS_DONE_OFFSET = 5;
+        final int ARRAY_OFFSET = 1;
+        final int TODO = 3;
+        final int EVENT = 2;
+        final int DEADLINE = 1;
```

Comment:
Avoid this approach. a small mistake in setting each of the variables to the correct numbers can lead to buggy code that is hard to spot. use enums